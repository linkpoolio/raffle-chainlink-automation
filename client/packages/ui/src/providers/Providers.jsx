import React from 'react'
import { Router } from 'react-router-dom'
import { Web3WalletProvider } from './'

export const Providers = ({ children, history }) => (
  <Router history={history}>
    <Web3WalletProvider>{children}</Web3WalletProvider>
  </Router>
)
