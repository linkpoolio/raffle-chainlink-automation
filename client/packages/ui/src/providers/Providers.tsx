import React from 'react'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
import {
  ConnectKitProvider,
  ConnectKitButton,
  getDefaultClient
} from 'connectkit'
import { WagmiConfig, createClient } from 'wagmi'

import { client } from '@ui/config/client'
import { Web3WalletProvider } from './'

const history = createBrowserHistory()

export const Providers = ({ children }) => (
  <Router history={history}>
    <WagmiConfig client={client}>
      <ConnectKitProvider>
        <Web3WalletProvider>{children}</Web3WalletProvider>
      </ConnectKitProvider>
    </WagmiConfig>
  </Router>
)