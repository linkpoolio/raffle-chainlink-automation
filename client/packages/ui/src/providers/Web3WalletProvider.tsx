import React from 'react'
import { ethers } from 'ethers'
import { WagmiConfig, createClient } from 'wagmi'
import { InjectedConnector } from 'wagmi/connectors/injected'
import { mainnet } from 'wagmi/chains'

const defaultChains = [mainnet]

const getProvider = () =>
  new ethers.providers.JsonRpcProvider('https://main-light.eth.linkpool.io', 1)

const connectors = () => {
  return [
    new InjectedConnector({
      chains: defaultChains,
      options: { shimDisconnect: true }
    })
  ]
}

const client = createClient({
  autoConnect: true,
  connectors,
  provider: getProvider
})

export const Web3WalletProvider = ({ children }) => (
  <WagmiConfig client={client}>{children}</WagmiConfig>
)
