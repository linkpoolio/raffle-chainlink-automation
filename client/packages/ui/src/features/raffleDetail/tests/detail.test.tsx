import React from 'react'
import renderer from 'react-test-renderer'

import { Providers } from '@ui/providers'
import { RaffleDetail } from '../'

const getComponent = () => {
  const component = (
    <Providers>
      <RaffleDetail id={1} />
    </Providers>
  )

  return component
}

describe('RaffleDetail', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(getComponent()).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
