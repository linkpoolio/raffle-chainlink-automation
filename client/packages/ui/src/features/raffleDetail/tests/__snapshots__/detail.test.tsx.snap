// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaffleDetail matches snapshot 1`] = `
[
  <div
    className="chakra-container css-1sgxzlz"
  >
    <div
      className="css-1tr0t8b"
    >
      <h2
        className="chakra-heading css-1jg9pmt"
      />
      <p
        className="chakra-text css-19rlujp"
      >
        View and manage raffle
      </p>
    </div>
    <div
      className="css-0"
    >
      <div
        className="css-z2pmpb"
      >
        <p
          className="chakra-text css-0"
        >
          Status
        </p>
        <p
          className="chakra-text css-0"
        >
          Finished
        </p>
      </div>
      <div
        className="css-z2pmpb"
      >
        <p
          className="chakra-text css-0"
        >
          Name
        </p>
        <p
          className="chakra-text css-0"
        />
      </div>
      <div
        className="css-z2pmpb"
      >
        <p
          className="chakra-text css-0"
        >
          Type
        </p>
        <p
          className="chakra-text css-0"
        >
          Dynamic
        </p>
      </div>
      <div
        className="css-z2pmpb"
      >
        <p
          className="chakra-text css-0"
        >
          Owner
        </p>
        <p
          className="chakra-text css-0"
        />
      </div>
      <div
        className="css-gmuwbf"
      >
        <div
          className="chakra-stack css-iew95a"
        />
      </div>
    </div>
  </div>,
  <span />,
  <span
    hidden={true}
    id="__chakra_env"
  />,
]
`;
